File: examples and misc/Java/PointersStrings.java
public class PointersStrings {
    public static void manipulateString(StringBuilder str) {
        for (int i = 0; i < str.length(); i++) {
            char c = str.charAt(i);
            if (c >= 'a' && c <= 'z') {
                str.setCharAt(i, (char) (c - 32)); // Convert to uppercase
            }
        }
    }

    public static void main(String[] args) {
        StringBuilder str = new StringBuilder("hello world");
        manipulateString(str);
        System.out.println("String after manipulation: " + str);
    }
}

File: examples and misc/C/pointers_strings.c
#include <stdio.h>
#include <ctype.h>

// Function to manipulate a string using pointers
void manipulate_string(char* str) {
    while (*str) {
        *str = toupper(*str);
        str++;
    }
}

int main() {
    char str[] = "hello world";
    manipulate_string(str);

    printf("String after manipulation: %s\n", str);
    return 0;
}
